File: project/jimmer-sql-test-model/src/main/java/org/babyfish/jimmer/sql/model/fetcher/Issue1434User.java
package org.babyfish.jimmer.sql.model.fetcher;

import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.Id;
import org.babyfish.jimmer.sql.JoinTable;
import org.babyfish.jimmer.sql.ManyToMany;
import org.babyfish.jimmer.sql.Table;

import java.util.List;

@Entity
@Table(name = "ISSUE_1434_USER")
public interface Issue1434User {

    @Id
    long id();

    String name();

    @ManyToMany
    @JoinTable(
            name = "ISSUE_1434_USER_DEPARTMENT_MAPPING",
            joinColumnName = "USER_ID",
            inverseJoinColumnName = "DEPARTMENT_ID"
    )
    List<Issue1434Department> departments();
}

File: project/jimmer-sql-kotlin-test-model/src/main/kotlin/org/babyfish/jimmer/sql/kt/model/fetcher/Issue1434User.kt
package org.babyfish.jimmer.sql.kt.model.fetcher

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.JoinTable
import org.babyfish.jimmer.sql.ManyToMany
import org.babyfish.jimmer.sql.Table

@Entity
@Table(name = "ISSUE_1434_USER")
interface Issue1434User {

    @Id
    val id: Long

    val name: String

    @ManyToMany
    @JoinTable(
        name = "ISSUE_1434_USER_DEPARTMENT_MAPPING",
        joinColumnName = "USER_ID",
        inverseJoinColumnName = "DEPARTMENT_ID"
    )
    val departments: List<Issue1434Department>
}
